<?xml version="1.0" ?>
<launch>    
    <arg name="use_imu" default="false"/>
    <arg name="use_lidar" default="false" />
    <arg name="config_path" default="$(find thormang3_p_manager)/config" />
    
    <param name="gazebo"                   value="true"    type="bool"/>
    <param name="gazebo_robot_name"        value="thormang3"/>
    
    <param name="offset_file_path"         value="$(arg config_path)/offset.yaml"/>
    <param name="robot_file_path"          value="$(arg config_path)/THORMANG3_P.robot"/>
    <param name="init_file_path"           value="$(arg config_path)/dxl_init.yaml"/>
        
    <param name="ft_data_path"             value="$(arg config_path)/ft_data.yaml"/>
    <param name="ft_calibration_data_path" value="$(arg config_path)/ft_calibration_data.yaml"/>

    <param name="action_file_path"         value="$(find thormang3_action_module)/data/motion_4096.bin"/>
                	 
    <!-- imu sensor package -->
    <include file="$(find thormang3_imu_3dm_gx4)/launch/imu.launch" if="$(arg use_imu)"/> 
    
    <!-- lidar -->    
    <include file="$(find thormang3_description)/launch/thor_laserscan.launch" if="$(arg use_lidar)"/> 
    
    <!-- THORMANG3_P Manager -->
    <node pkg="thormang3_p_manager" type="thormang3_p_manager" name="thormang3_p_manager" output="screen"/>
    
    <!-- Robot Model & TF -->
    <include file="$(find thormang3_description)/launch/thormang3_mpc.launch"/> 
</launch>
